matplotcheck/raster.py

```diff
@@ -140,25 +140,36 @@ def assert_legend_accuracy_classified_image(
         # Check that each legend entry label is in one of all_label_options
         assert len([val for val in legend_dict.values() if val]) == len(
             all_label_options
-        ), "Incorrect legend labels"
+        ), (
+            "Number of label options provided doesn't match the number of",
```

Comment:
@nkorinek i can't comment inline here but i'd like to see this method broken up into pieces we can use and test independently.

1. get_plot_image or get_image. i suspect this method already exists and we can just reuse it here to grab the array rather than rewriting it which creates redundancy in the code
2. get_legend labels -- that could be a nice helper that we can look at and test.
3. finally the assert itself which compares an input list to the expected values.